test/unit/services/test_manufacturer.py

```diff
@@ -2,237 +2,382 @@
+from test.unit.services.conftest import ServiceTestHelpers
 
 import pytest
 from bson import ObjectId
 
+from inventory_management_system_api.core.custom_object_id import CustomObjectId
 from inventory_management_system_api.core.exceptions import MissingRecordError
 from inventory_management_system_api.models.manufacturer import ManufacturerIn, ManufacturerOut
-from inventory_management_system_api.schemas.manufacturer import (
-    AddressSchema,
-    ManufacturerPatchSchema,
-    ManufacturerPostSchema,
-)
-
-
-def test_create(
-    test_helpers,
-    manufacturer_repository_mock,
-    model_mixins_datetime_now_mock,  # pylint: disable=unused-argument
-    manufacturer_service,
-):
-    """
-    Testing creating a manufacturer
-    """
-    # pylint: disable=duplicate-code
-
-    manufacturer = ManufacturerOut(
-        _id=str(ObjectId()),
-        name="Manufacturer A",
-        code="manufacturer-a",
-        url="http://testUrl.co.uk",
-        address=AddressSchema(
-            address_line="1 Example Street",
-            town="Oxford",
-            county="Oxfordshire",
-            postcode="OX1 2AB",
-            country="United Kingdom",
-        ),
-        telephone="0932348348",
-        created_time=MODEL_MIXINS_FIXED_DATETIME_NOW,
-        modified_time=MODEL_MIXINS_FIXED_DATETIME_NOW,
-    )
-    # pylint: enable=duplicate-code
-
-    # Mock `create` to return the created manufacturer
-    test_helpers.mock_create(manufacturer_repository_mock, manufacturer)
-
-    created_manufacturer = manufacturer_service.create(
-        ManufacturerPostSchema(
-            name=manufacturer.name,
-            url=manufacturer.url,
-            address=manufacturer.address,
-            telephone=manufacturer.telephone,
+from inventory_management_system_api.schemas.manufacturer import ManufacturerPatchSchema, ManufacturerPostSchema
+from inventory_management_system_api.services import utils
+from inventory_management_system_api.services.manufacturer import ManufacturerService
+
+
+class ManufacturerServiceDSL:
+    """
+    Base class for `ManufacturerService` unit tests.
+    """
+
+    wrapped_utils: Mock
+    mock_manufacturer_repository: Mock
+    manufacturer_service: ManufacturerService
+
+    @pytest.fixture(autouse=True)
+    def setup(
+        self,
+        manufacturer_repository_mock,
+        manufacturer_service,
+        # Ensures all created and modified times are mocked throughout
+        # pylint: disable=unused-argument
+        model_mixins_datetime_now_mock,
+    ):
+        """Setup fixtures"""
+        self.mock_manufacturer_repository = manufacturer_repository_mock
+        self.manufacturer_service = manufacturer_service
+
+        with patch("inventory_management_system_api.services.manufacturer.utils", wraps=utils) as wrapped_utils:
+            self.wrapped_utils = wrapped_utils
+            yield
+
+
+class CreateDSL(ManufacturerServiceDSL):
+    """
+    Base class for create tests.
```

Comment:
```suggestion
    Base class for `create` tests.
```

Reply:
Shouldn't we do this in the the repo unit tests as well?

Reply:
I did it for them as well as they weren't like that.